src/input/input_python.rs

```diff
@@ -302,35 +302,40 @@ impl<'a> Input<'a> for PyAny {
         }
     }
 
-    fn ultra_strict_float(&self) -> ValResult<f64> {
+    fn ultra_strict_float(&'a self) -> ValResult<EitherFloat<'a>> {
         if self.is_instance_of::<PyInt>() {
             Err(ValError::new(ErrorType::FloatType, self))
         } else if let Ok(float) = self.extract::<f64>() {
```

Comment:
we shouldn't use extract here, but rather `is_instance_of`, then use `EitherFloat::Py`.